### PR TITLE
test(dtslint): add dtslint test for merge operator (#4093)

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:** Add dtslint test for merge operator  

**Related issue (if exists):** #4093

### CODE REVIEW

File: spec-dtslint/operators/merge-spec.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+});
+
+it('should infer correctly with concurrent and scheduler param', () => {
+  const a = of(1, 2, 3).pipe(merge(3, asyncScheduler)); // $ExpectType Observable<number>
+});
+
+it('should infer correctly with 1 Observable param', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const res = a.pipe(merge(b)); // $ExoectType Observable<number|string>
```
Comment (benlesh) [inline spec-dtslint/operators/merge-spec.ts]: Typos throughout this PR: `ExpoectType` => `ExpectType` 

:)

Comment (dkosasih) [reply spec-dtslint/operators/merge-spec.ts]: Opps... sorry @benlesh 
Fixed now.

**Comment by benlesh** [comment]
@dkosasih it looks like the formats of your expectations are wrong, travis is failing on the test: https://travis-ci.org/ReactiveX/rxjs/jobs/476546817

**Comment by dkosasih** [comment]
I'm pretty sure it was working 🤦‍♂️ ...

File: spec-dtslint/operators/merge-spec.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of('d', 'e', 'f');
+  const d = of('g', 'h', 'i');
+  const e = of('j', 'k', 'l');
+  const f = of('m', 'n', 'o');
+  const res = a.pipe(merge(b, c, d, e, f, 1, asyncScheduler)); // $ExpectType Observable<string | number>
+});
+
+// TODO: Fix this when the both merge operator and merge creator function has been fix
```
Comment (kellyprankin) [inline spec-dtslint/operators/merge-spec.ts]: Is this resolved?

**Comment by dkosasih** [comment]
@cartant 
Changed the test to use helpers from Ben's PR

**Pull request merged**